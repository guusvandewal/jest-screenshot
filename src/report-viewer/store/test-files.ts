import { component } from "tsdi";
import { observable, action, computed } from "mobx";
import { FileReport } from "../../reporter-types";

@component
export class StoreTestFiles {
    @observable public activeFileName = testResults.files.length > 0 ? testResults.files[0].testFilePath : undefined;

    @action.bound public selectFile(file: FileReport) {
        this.activeFileName = file.testFilePath;
    }

    @computed public get activeFile() {
        return testResults.files.find(file => file.testFilePath === this.activeFileName);
    }

    public isActive(file: FileReport) {
        return this.activeFileName === file.testFilePath;
    }
}
