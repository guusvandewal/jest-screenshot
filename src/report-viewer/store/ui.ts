import { component } from "tsdi";
import { observable, action, computed } from "mobx";
import { FileReport } from "../../reporter-types";
import { bind } from "lodash-decorators";

@component
export class StoreUi {
    @observable public menuVisible = true;

    @bind @action public toggleMenu() {
        this.menuVisible = !this.menuVisible;
    }
}
