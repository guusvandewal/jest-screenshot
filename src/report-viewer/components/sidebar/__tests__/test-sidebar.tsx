import * as React from "react";
import { shallow } from "enzyme";
import { Sidebar } from "..";
import { StoreUi } from "../../../store";

describe("Sidebar", () => {
    it("looks as expected", () => {
        expect(shallow(<Sidebar />)).toMatchSnapshot();
    });

    it("isn't rendered with the sidebar hidden", () => {
        tsdi.get(StoreUi).toggleMenu();
        expect(shallow(<Sidebar />)).toMatchSnapshot();
    });
});
