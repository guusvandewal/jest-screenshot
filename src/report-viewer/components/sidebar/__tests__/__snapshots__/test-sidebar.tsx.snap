// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar isn't rendered with the sidebar hidden 1`] = `""`;

exports[`Sidebar looks as expected 1`] = `
<aside
  className="column is-3 section is-fullheight menu"
>
  <p
    className="menu-label"
  />
  <FileList />
</aside>
`;
